hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Sunflower IT <http://sunflowerweb.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import timedelta
+
+from odoo import models, fields, api, _
+from odoo.exceptions import Warning as UserError
+
+
+class HrHolidays(models.Model):
+    """Update analytic lines on status change of Leave Request"""
+    _inherit = 'hr.holidays'
+
+    # Timesheet entry linked to this leave request
+    analytic_line_ids = fields.One2many(
+        'account.analytic.line',
+        'leave_id',
+        'Analytic Lines'
+    )
+
+    @api.multi
+    def add_timesheet_line(self, description, date, hours, account_id,
+                           user_id, sheet_id):
+        """Add a timesheet line for this leave"""
+        self.ensure_one()
+        self.sudo(user_id).write({'analytic_line_ids': [(0, False, {
+            'name': description,
+            'date': date,
+            'unit_amount': hours,
+            'company_id': self.employee_id.company_id.id,
+            'account_id': account_id,
+            'sheet_id': sheet_id,
+        })]})
+
+    @api.model
+    def _get_hours_per_day(self, company, employee):
+        """Can be overridden to consider employee details also"""
+        hours_per_day = company.timesheet_hours_per_day
+        if not hours_per_day:
+            raise UserError(
+                _("No hours per day defined for Company '%s'") %
+                (company.name,))
+        return hours_per_day
+
+    @api.model
+    def _find_sheet(self, employee, dt_current):
```

Comment:
@aheficent Probably here to add a condition about states of the sheet, just like in _compute_sheet from account.analytic.line, https://github.com/odoo/odoo/blob/10.0/addons/hr_timesheet_sheet/models/account_analytic_line.py#L15